graphql/src/schema/api.rs

```diff
@@ -212,19 +212,42 @@ fn field_scalar_filter_input_values(
     field: &Field,
     field_type: &ScalarType,
 ) -> Vec<InputValue> {
-    vec![
-        Some(input_value(
-            &field.name,
+    match field_type.name.as_ref() {
+        "BigInt" => vec!["", "not", "gt", "lt", "gte", "lte", "in", "not_in"],
+        "Boolean" => vec!["", "not", "in", "not_in"],
+        "Bytes" => vec!["", "not", "in", "not_in", "contains", "not_contains"],
+        "Float" => vec!["", "not", "gt", "lt", "gte", "lte", "in", "not_in"],
+        "ID" => vec!["", "not", "gt", "lt", "gte", "lte", "in", "not_in"],
+        "Int" => vec!["", "not", "gt", "lt", "gte", "lte", "in", "not_in"],
+        "List" => vec!["", "not", "in", "not_in", "contains", "not_contains"],
+        "String" => vec![
             "",
-            Type::NamedType(field_type.name.to_owned()),
-        )),
-        Some(input_value(
-            &field.name,
             "not",
-            Type::NamedType(field_type.name.to_owned()),
-        )),
-    ].into_iter()
-        .filter_map(|value_opt| value_opt)
+            "gt",
+            "lt",
+            "gte",
+            "lte",
+            "in",
+            "not_in",
+            "contains",
+            "not_contains",
+            "starts_with",
+            "not_starts_with",
+            "ends_with",
+            "not_ends_with",
+        ],
+        _ => vec!["", "not"],
+    }.into_iter()
+        .map(|filter_type| {
+            let field_type = Type::NamedType(field_type.name.to_owned());
+            let value_type = match filter_type {
+                "in" | "not_in" => {
+                    Type::ListType(Box::new(Type::NonNullType(Box::new(field_type))))
```

Comment:
I _think_ there needs to be another `Type::NonNullType` around the `Type::ListType` to disallow `null` being passed in as a value (anything from an empty list upwards is fine, but `null` makes no sense).